### PR TITLE
Support streaming in size estimation function in `push_to_hub`

### PR DESCRIPTION
**Pull request by mariosasko**
This PR adds the streamable version of `os.path.getsize` (`fsspec` can return `None`, so we fall back to `fs.open` to make it more robust) to account for possible streamable paths in the nested `extra_nbytes_visitor` function inside `push_to_hub`.

### CODE REVIEW

File: src/datasets/utils/streaming_download_manager.py [hunk 1]
```diff
@@ -215,6 +215,33 @@ def xisfile(path, use_auth_token: Optional[Union[str, bool]] = None) -> bool:
...
+        if rest_hops and fsspec.get_fs_token_paths(rest_hops[0])[0].protocol == "https":
+            storage_options = {
+                "https": {"headers": get_authentication_headers_for_url(rest_hops[0], use_auth_token=use_auth_token)}
+            }
+        else:
+            storage_options = None
+        fs, *_ = fsspec.get_fs_token_paths(path, storage_options=storage_options)
+        size = fs.size(main_hop)
+        if size is None:
+            with fs.open(main_hop, "rb") as f:
```
[the remark this answers is not in the document]
Comment (lhoestq) [reply src/datasets/utils/streaming_download_manager.py:240]: Yes I think so

**Comment by severo** [comment]
would this allow to include the size in the dataset info without downloading the files? related to https://github.com/huggingface/datasets/pull/3670

**Comment by mariosasko** [comment]
@severo I don't think so. We could use this to get `info.download_checksums["num_bytes"]`, but we must process the files to get the rest of the size info.

**Review by lhoestq — APPROVED** [review]
LGTM :)

**Pull request merged**